app/ts/simulation/services/EthereumClientService.ts

```diff
@@ -137,6 +137,20 @@ export class EthereumClientService {
 		return EthereumBlockHeader.parse(await this.requestHandler.jsonRpcRequest({ method: 'eth_getBlockByNumber', params: [blockTag, fullObjects] }))
 	}
 
+	public async getBlockByHash(blockHash: EthereumBytes32, fullObjects: boolean = true): Promise<EthereumBlockHeaderWithTransactionHashes | EthereumBlockHeader> {
+		const cached = this.getCachedBlock()
+		if (cached && (cached.hash === blockHash)) {
+			if (fullObjects === false) {
+				return { ...cached, transactions: cached.transactions.map((transaction) => transaction.hash) }
+			}
+			return cached
+		}
+		if (fullObjects === false) {
+			return EthereumBlockHeaderWithTransactionHashes.parse(await this.requestHandler.jsonRpcRequest({ method: 'eth_getBlockByHash', params: [blockHash, false] }))
```

Comment:
Is it possible for the RPC to return a response that doesn't parse the funtypes validation? safeParse case instead?

Reply:
yeah, the RPC server can return whatever, I feel it should be error thought?